perfkitbenchmarker/benchmarks/cloudsuite_web_search.py

```diff
@@ -0,0 +1,219 @@
+
+FABAN_OUTPUT_DIR = posixpath.join(vm_util.VM_TMP_DIR, 'outputFaban')
+
+PACKAGES_URL = ('http://parsa.epfl.ch/cloudsuite/software/perfkit/web_search')
+NUTCH_SITE_URL = posixpath.join(PACKAGES_URL, 'nutch-site.xml')
+CRAWLED_URL = posixpath.join(PACKAGES_URL, 'crawl.tar.gz')
+SCHEMA_URL = posixpath.join(PACKAGES_URL, 'schema.xml')
+SEARCH_DRIVER_URL = posixpath.join(PACKAGES_URL, 'search.tar.gz')
+
+SOLR_PORT = 8983
+
+fw = None
+
+
+def GetInfo():
+  return BENCHMARK_INFO
+
+
+def _PrepareSolr(solr_nodes):
+  """Starts and configures SolrCloud."""
+  stdout, _ = solr_nodes[0].RemoteCommand('cd {0} && '
+                                          'wget --quiet {1} && '
+                                          'cat schema.xml'.format(
+                                              CLOUDSUITE_WEB_SEARCH_DIR,
+                                              SCHEMA_URL))
+  for vm in solr_nodes:
+    vm.Install('solr521')
+    solr.SetSchema(vm, stdout.replace('"', '\\"'))
+    if vm == solr_nodes[0]:
+      solr.StartWithZookeeper(vm, fw, SOLR_PORT)
+    else:
+      solr.Start(vm, fw, SOLR_PORT, solr_nodes[0], SOLR_PORT + 1000)
+  solr.CreateCollection(solr_nodes[0], 'cloudsuite_web_search', len(solr_nodes))
+
+
+def _BuildIndex(indexer, solr_node):
+  """Downloads data and builds Solr index from it."""
+  indexer.Install('nutch110')
+  indexer.RemoteCommand('cd {0} && '
+                        'wget {1} && '
+                        'sed -i "/<value>http/c\\<value>http://{2}:'
+                        '{3}/solr/cloudsuite_web_search</value>" '
+                        'nutch-site.xml'.format(
+                            CLOUDSUITE_WEB_SEARCH_DIR,
+                            NUTCH_SITE_URL, solr_node.ip_address,
+                            SOLR_PORT))
+  stdout, _ = indexer.RemoteCommand('cd {0} && '
+                                    'cat nutch-site.xml'.format(
+                                        CLOUDSUITE_WEB_SEARCH_DIR))
+  nutch.ConfigureNutchSite(indexer, stdout.replace('"', '\\"'), solr_node,
+                           SOLR_PORT, 'cloudsuite_web_search')
+  indexer.RobustRemoteCommand('cd {0}/runtime/local && '
+                              'wget {1}  && '
+                              'export JAVA_HOME={2} && '
+                              'tar -zxf crawl.tar.gz && '
+                              'bin/nutch index crawl/crawldb/ -linkdb '
+                              'crawl/linkdb/ -dir crawl/segments/'.format(
+                                  nutch.NUTCH_HOME_DIR, CRAWLED_URL,
+                                  nutch.java_home))
+
+
+def _PrepareClient(client, solr_nodes):
+  """Prepares client machine in the benchmark."""
+  client.Install('faban')
+  faban.Start(client, fw)
+  client.RemoteCommand('cd {0} && '
+                       'wget {1} && '
+                       'tar -xzf search.tar.gz && '
+                       'sed -i "/faban.home/c\\faban.home={0}" '
```

Comment:
Would it be possible to break this up a bit? Maybe a command for each configuration file you're changing?